spec/integration/util/windows/adsi_spec.rb

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env ruby
+require 'spec_helper'
+require 'puppet/util/windows'
+
+def fail_if_utf8_codepage
+  if Puppet.features.microsoft_windows? && Encoding.default_external == Encoding::UTF_8
+    raise 'This test must be run in a codepage other than 65001 to validate behavior'
```

Comment:
Yeah, we have a few other tests that do the same thing - and perhaps should be rewritten.

The problem is that we can't validate the "fixed" behavior unless we're in another codepage.